### PR TITLE
Move options subsystem over to using SolutionService snapshots.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/63241

This is continued work toward banning getting back to the Workspace from certain layers of Roslyn (like the features/code_style layers).

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Options/OptionsProvider.cs [hunk 1]
```diff
@@ -10,5 +10,5 @@ namespace Microsoft.CodeAnalysis;
 
 internal interface OptionsProvider<TOptions>
 {
-    ValueTask<TOptions> GetOptionsAsync(HostLanguageServices languageServices, CancellationToken cancellationToken);
+    ValueTask<TOptions> GetOptionsAsync(Host.LanguageServices languageServices, CancellationToken cancellationToken);
```
Comment (tmat) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Options/OptionsProvider.cs:13]: >Host [](http://example.com/codeflow?start=40&length=4)

nit: `Host.` seems unnecessary

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/CodeActionOptions.cs [hunk 2]
```diff
@@ -97,48 +97,48 @@ internal interface CodeActionOptionsProvider :
         SimplifierOptionsProvider,
         AddImportPlacementOptionsProvider
     {
-        CodeActionOptions GetOptions(HostLanguageServices languageService);
+        CodeActionOptions GetOptions(Host.LanguageServices languageService);
```
Comment (tmat) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/CodeActionOptions.cs:100]: >Host. [](http://example.com/codeflow?start=37&length=5)

`Host.` unnecessary

**Comment by tmat** [comment]
>     private readonly HostLanguageServices _languageServices;

ProjectServices?

---
Refers to: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeActions/CodeFixOptionsProvider.cs:33 in 5ed6908. [](commit_id = 5ed69083ffc287f48ba259262b6612fb65bae4cb, deletion_comment = False)

**Comment by CyrusNajmabadi** [comment]
@tmat  please re-review.  Note the `Host.` prefix will be going away in next PR.  thanks!

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/CodeActionOptions.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/CodeActionOptions.cs:100]: Taken care of in next PR.

**Comment by CyrusNajmabadi** [comment]
@tmat please re-review. Note the Host. prefix will be going away in next PR. thanks!

**Review by tmat — APPROVED** [review]
:shipit:

**Pull request merged**